packages/widget-chat/src/containers/message-composer/index.test.js

```diff
@@ -1,22 +1,115 @@
+/* eslint-disable max-nested-callbacks */
 import React from 'react';
 import renderer from 'react-test-renderer';
 
-import MessageComposer from '.';
+import {Provider} from 'react-redux';
+import {applyMiddleware, combineReducers, compose, createStore} from 'redux';
+import thunk from 'redux-thunk';
+
+import {user, conversation, message} from '../../reducers';
+import sparkReducer from '../../modules/redux-spark/reducers';
+
+
+import ConnectedMessageComposer, {MessageComposer} from '.';
+let component, store;
 
 describe(`MessageComposer component`, () => {
-  const component = renderer.create(
-    <MessageComposer
-      placeholder="Message Placeholder"
-      value="This is a message"
-    />
-  );
-
-  it(`renders properly`, () => {
-    expect(component).toMatchSnapshot();
+  beforeEach(() => {
+    store = createStore(
+      combineReducers({
+        user,
+        conversation,
+        message,
+        spark: sparkReducer
+      }),
+      compose(applyMiddleware(thunk))
+    );
+
+    component = renderer.create(
```

Comment:
I don't think any of the tests directly mutate `component`? If that's the case, we probably don't need to re-build the component after each tests.